clowdapp.yaml

```diff
@@ -12,8 +12,13 @@ objects:
       ${LINT_ANNOTATION}: "minimum three replicas not required"
   spec:
     envName: ${ENV_NAME}
-    dependencies: []
-    optionalDependencies: []
+    testing:
+      iqePlugin: xjoin
+    dependencies:
+      - xjoin-api-gateway
```

Comment:
Please note that this is a self-sufficient service that does not require dependencies. With regards to testing, there can be a smoke test on checking that the service runs and maps clowder config correctly. cc @skateman to keep me honest.

Reply:
True story

Reply:
I added `xjoin-api-gateway` to the dependencies because when the dependencies/optionalDependencies were last reimplemented, we were told that anything that is required to run from the end-user/customer point of view should be considered hard dependency. Here, a customer will not use `xjoin-apicurio` on it's own, but only through `xjoin-api-gateway`.

This is the same reason why `ingress` is a hard dependency of `host-inventory`. Host Inventory can 100% function properly without `ingress`, theoretically, any other service could send a Kafka message to Inventory and create a new host (and there are other reporters), but we were told that we should always deploy the entire end-to-end pipeline together.

Here is the doc I'm talking about: https://docs.google.com/document/d/1tOwLbKb7ZMYLUpo8K1Wn-iNIoJOZWfrz1OPWDbtcrJc/edit#heading=h.lvl0oehd4hnk

Reply:
The direction of dependency tree should be reversed here. apicurio is required by xjoin-api-gateway. 

This is a support service (like a database) that can be offloaded to a managed service in future. The goal of the testing should be around the wrapper that we're adding (that env var mapping), making sure that it works with clowder.

Reply:
Yeah, it just requires a simple test to make sure that the service is up and running and responds to simple requests.